zer0-os/zOS
# feat: web3 login

[JIRA](https://wilderworld.atlassian.net/browse/ZOS-293)

### What does this do?

Adds ability to sign in with web3.

Also some refactoring around `Login` to make "choose sign-in option" logic simpler, and to make it easier to add more options.

### Why are we making this change?

Need to be able to sign in with web3.

### How do I test this?

1. Open `/login` endpoint
2. Hit the `Web3 Wallet` button
3. Select your wallet of choice
4. Reject transaction, check that the error message shows up
5. Return to `Email` login, return to `Web3 Wallet` login, check error message is gone
6. Select your wallet of choice
7. Sign transaction
8. You should now be logged in

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?
